### PR TITLE
ExampleSand & ExampleCactus

### PR DESCRIPTION
**Pull request by GoodPro712**
### Description of the Change

Added an example sand tile, the projectile that spawns when the tile below it gets destroyed, then an example sand block item to place the sand tile

### Alternate designs

N/A

### Why this should be merged into tModLoader

This should be merged into tModLoader because there currently is no example falling block meaning its a lot harder for modders to make such a thing

### Benefits

The benefits from an ExampleSand is that modders have an easier way of knowing how to make a falling block

### Possible drawbacks

N/A

### Applicable Issues

I haven't added comments.

### Sample Usage

N/A

### Notes
This code was originally from the Terraria source code then blushie's mod (I asked him and he said we could use it) used it for a sand block, then I modified it for examplemod

### CODE REVIEW

File: ExampleMod/ExampleWorld.cs [hunk 1]
```diff
@@ -162,14 +162,14 @@ public class ExampleWorld : ModWorld
...
-			for (int k = 0; k < (int)((double)(Main.maxTilesX * Main.maxTilesY) * 6E-05); k++) {
+			for (int k = 0; k < (int)((Main.maxTilesX * Main.maxTilesY) * 6E-05); k++) {
 				// The inside of this for loop corresponds to one single splotch of our Ore.
 				// First, we randomly choose any coordinate in the world by choosing a random x and y value.
 				int x = WorldGen.genRand.Next(0, Main.maxTilesX);
 				int y = WorldGen.genRand.Next((int)WorldGen.worldSurfaceLow, Main.maxTilesY); // WorldGen.worldSurfaceLow is actually the highest surface tile. In practice you might want to use WorldGen.rockLayer or other WorldGen values.
 
 				// Then, we call WorldGen.TileRunner with random "strength" and random "steps", as well as the Tile we wish to place. Feel free to experiment with strength and step to see the shape they generate.
-				WorldGen.TileRunner(x, y, (double)WorldGen.genRand.Next(3, 6), WorldGen.genRand.Next(2, 6), TileType<ExampleOre>(), false, 0f, 0f, false, true);
+				WorldGen.TileRunner(x, y, WorldGen.genRand.Next(3, 6), WorldGen.genRand.Next(2, 6), TileType<ExampleOre>());
```
Comment (GoodPro712) [inline ExampleMod/ExampleWorld.cs]: The cast wasn't doing anything.

File: ExampleMod/ExampleWorld.cs [hunk 2]
```diff
@@ -162,14 +162,14 @@ public class ExampleWorld : ModWorld
 
 			// Ores are quite simple, we simply use a for loop and the WorldGen.TileRunner to place splotches of the specified Tile in the world.
 			// "6E-05" is "scientific notation". It simply means 0.00006 but in some ways is easier to read.
-			for (int k = 0; k < (int)((double)(Main.maxTilesX * Main.maxTilesY) * 6E-05); k++) {
+			for (int k = 0; k < (int)((Main.maxTilesX * Main.maxTilesY) * 6E-05); k++) {
```
Comment (GoodPro712) [inline ExampleMod/ExampleWorld.cs]: The cast wasn't doing anything.

**Comment by GoodPro712** [comment]
woops

**Comment by JavidPack** [comment]
Great job, thanks for spearheading this.

**Comment by GoodPro712** [comment]
no problem

**Pull request merged**